models/api-models.js

```diff
@@ -67,4 +67,35 @@ exports.readCommentsByArticleId = (article_id) => {
          } 
          return rows;
     })
-}
\ No newline at end of file
+}
+
+exports.updateVotesByArticle = (inc_votes, article_id) => {
+
+
+    if(typeof inc_votes !== 'number' || isNaN(inc_votes)){
+        return Promise.reject({ status: 400, msg: 'inc_votes should be a number' })
+    }
+
+    return db.query(`
+        SELECT votes 
+        FROM articles
+        WHERE article_id = $1`, [article_id])
+        .then(({rows}) => {
+            if(rows.length === 0) {
+                return Promise.reject({ status: 404, msg: 'article not found'})
+            } 
+            
```

Comment:
You don't need to manually check the above - because article_id is a foreign key when you try to update and use a value that it cannot find - psql will throw an error itself. See what happens if you remove this check and whether you get an error passed to your catch block!